test/functional/app/api/test_user_api.py

```diff
@@ -1,4 +1,49 @@
 import pytest, json
+from faker.providers import phone_number
+from faker import Faker
+
+from server.utils.auth import get_complete_auth_token
+from server.utils.phone import proccess_phone_number
+
+fake = Faker()
+fake.add_provider(phone_number)
+
+
+@pytest.mark.parametrize("user_id_accessor, is_vendor, is_groupaccount, tier, status_code", [
+    (lambda o: o.id, False, False, 'subadmin', 403),
+    (lambda o: o.id, True, False, 'admin', 200),
```

Comment:
I like having tests for tiers in the api, as it will let us know if someone changes `@requires_auth(allowed_roles={'ADMIN': 'admin'})`. makes it more explicit